crates/electrum/src/electrum_ext.rs

```diff
@@ -139,59 +139,47 @@ pub trait ElectrumExt {
     ///
     /// - `prev_tip`: the most recent blockchain tip present locally
     /// - `keychain_spks`: keychains that we want to scan transactions for
-    /// - `txids`: transactions for which we want updated [`Anchor`]s
-    /// - `outpoints`: transactions associated with these outpoints (residing, spending) that we
-    ///     want to included in the update
-    ///
     /// The scan for each keychain stops after a gap of `stop_gap` script pubkeys with no associated
     /// transactions. `batch_size` specifies the max number of script pubkeys to request for in a
     /// single batch request.
-    fn scan<K: Ord + Clone>(
+    /// ##### NOTE: Scanning with keychain is very inefficient and should be used only when restoring
+    /// from seed words.
+    fn scan_with_keychain<K: Ord + Clone>(
         &self,
         prev_tip: Option<CheckPoint>,
         keychain_spks: BTreeMap<K, impl IntoIterator<Item = (u32, ScriptBuf)>>,
-        txids: impl IntoIterator<Item = Txid>,
-        outpoints: impl IntoIterator<Item = OutPoint>,
         stop_gap: usize,
         batch_size: usize,
     ) -> Result<(ElectrumUpdate, BTreeMap<K, u32>), Error>;
 
     /// Convenience method to call [`scan`] without requiring a keychain.
     ///
     /// [`scan`]: ElectrumExt::scan
-    fn scan_without_keychain(
+    fn sync(
         &self,
         prev_tip: Option<CheckPoint>,
         misc_spks: impl IntoIterator<Item = ScriptBuf>,
-        txids: impl IntoIterator<Item = Txid>,
-        outpoints: impl IntoIterator<Item = OutPoint>,
+        _txids: impl IntoIterator<Item = Txid>,
+        _outpoints: impl IntoIterator<Item = OutPoint>,
         batch_size: usize,
     ) -> Result<ElectrumUpdate, Error> {
         let spk_iter = misc_spks
             .into_iter()
             .enumerate()
             .map(|(i, spk)| (i as u32, spk));
 
-        let (electrum_update, _) = self.scan(
-            prev_tip,
-            [((), spk_iter)].into(),
-            txids,
-            outpoints,
-            usize::MAX,
-            batch_size,
-        )?;
+        let (electrum_update, _) =
+            self.scan_with_keychain(prev_tip, [((), spk_iter)].into(), usize::MAX, batch_size)?;
 
         Ok(electrum_update)
     }
 }
 
 impl ElectrumExt for Client {
-    fn scan<K: Ord + Clone>(
+    fn scan_with_keychain<K: Ord + Clone>(
         &self,
         prev_tip: Option<CheckPoint>,
         keychain_spks: BTreeMap<K, impl IntoIterator<Item = (u32, ScriptBuf)>>,
-        txids: impl IntoIterator<Item = Txid>,
-        outpoints: impl IntoIterator<Item = OutPoint>,
```

Comment:
Now that we are removing `txids` and `outpoints` it doesn't make sense to be calling `populate_with_txids` and `populate_with_outpoints` here. Should separate methods be created to get these updates? @LLFourn @evanlinjin

Reply:
Yes the separate method is `scan`. `scan` is still meant to be doing this.